### PR TITLE
17644 Update Destination data type test to use the new data types

### LINKED ISSUE #17644: Update the destination data type tests to use the new  data types

### PR DESCRIPTION
**Pull request by suhomud**
## What
Prerequisite :  [see](https://docs.google.com/document/d/1aJdnNgpqm6rVV711HaABEpCz6wdl3deCA7A41Rl4CRI/edit#heading=h.a98kcaptkze3)
<p>

Add support of new protocol versions for `DestinationAcceptanceTest`. With this change we can handle different protocol versions in destination integration tests.
## How
Added new method `getProtocolVersion()` to `DestinationAcceptanceTest`  which can be overridden in child classes.  
original method example: 
```
  public ProtocolVersion getProtocolVersion() {
    return ProtocolVersion.DEFAULT;
  }
```
overridden:
 ```
 public ProtocolVersion getProtocolVersion() {
    return ProtocolVersion.V0;
  }
```
The ProtocolVersion is used to prefix dataset files 
Screenshot 2022-11-11 at 02 33 04

## Recommended reading order
1. `DestinationAcceptanceTest` - new getProtocolVersion method, prefixing files name.
2. `ProtocolVersion` - enum with supported protocol version and file prefix 
3.  `ArgumentProviderUtil` prefix file name method, reflection magic (This one is used to get protocol version in Argument Providers, since during it's execution test Instance is not instantiated. PS: Feel free to argue this approach since this result can be achieved  in a tests itself without modifying  Argument Providers )
4. `DataArgumentsProvider`
5. `DataTypeTestArgumentProvider`
6. `NamespaceTestCaseProvider` moved from `DestinationAcceptanceTest`.
7. `NumberDataTypeTestArgumentProvider`
8.  all dataset files under `resources/v0` folder

### CODE REVIEW

**Comment by suhomud** [comment]
Tests result above 👆👆👆 has no changes and it's uptodate with master branch

**Comment by suhomud** [comment]
Test result below 👇👇👇 after change

**Review by grishick — COMMENTED** [review]
nit: as far as I can see `NamespaceTestCaseProvider` is still used only inside `DestinationAcceptanceTest`, so I don't see a reason to make it into a separate class at this point. Otherwise, looks good.

**Review by grishick — APPROVED** [review]
LGTM, but @edgao should review as well before this can be merged

**Review by edgao — COMMENTED** [review]
this looks great! just a few comments 🚛

File: airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/ProtocolVersion.java
Comment (suhomud) [reply airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/ProtocolVersion.java:10]: Nice! I was confused with DEFAULT. With v0 and v1 it will be much clearer. Thanks!

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/v0/data_type_array_object_test_catalog.json
Comment (suhomud) [reply airbyte-integrations/bases/standard-destination-test/src/main/resources/v0/data_type_array_object_test_catalog.json:31]: Got it. Removed it from data_type_array_object_test_catalog.json and data_type_array_object_test_catalog.txt

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/v0/data_type_array_object_test_catalog.json
Comment (edgao) [reply airbyte-integrations/bases/standard-destination-test/src/main/resources/v0/data_type_array_object_test_catalog.json]: sorry, I meant that all the test files should include binary data (i.e. also data_type_array_test, data_type_basic_test, etc)

**Review by edgao — APPROVED** [review]
just one comment, but otherwise this looks good!

**Review by edgao — COMMENTED** [review]
derp. I also forgot that we should have tests for `Infinity`, `-Infinity`, and `NaN` number/integer values. (these should be added to the various `messages.txt` files)

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_object_test_catalog.json
Comment (etsybaev) [inline airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_object_test_catalog.json:18]: I see we also have TimeWithTimezone and TimeWithoutTimezone. Do we cover it?

Comment (suhomud) [reply airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_object_test_catalog.json:18]: Adding it. Thanks for noticing

**Comment by grishick** [comment]
This PR only changes tests and does not affect production, so it is safe to merge.

**Pull request merged**
